test/e2e/llmisvc/test_prestop_hook.py

```
@@ -0,0 +1,174 @@
+            should_skip_deletion = skip_all_deletion or (
+                skip_deletion_on_failure and test_failed
+            )
+
+            if not should_skip_deletion:
+                delete_llmisvc(kserve_client, test_case.llm_service)
+            elif test_failed and skip_deletion_on_failure:
+                print(
+                    f"⏭️  Skipping deletion of {service_name} due to test failure "
+                    f"(SKIP_DELETION_ON_FAILURE=True)"
+                )
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)